### PR TITLE
Params name not matching, caused first website granted not working

### LINKED ISSUE #19574: "Invite user" feature grants access to the default website, ignoring the selected website
When inviting a user using "Invite user" feature, and selecting a website (say ID=2) in the list to grant permission,
then after clicking `INVITE USER` button,
we can see that the user is granted the View permission to the first website in the list, rather than the website selected.

This is a security issue and should block release until fixed.

### PR DESCRIPTION
**Pull request by peterhashair**
### Description:

Fixes: #19574

This was caused by the param name, update it works now.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Review by bx80 — CHANGES_REQUESTED** [review]
I've recreated the issue prior to this fix, applied the fix and the functional issue is resolved.

I think it would be good to add a unit test where we call `inviteUser` with specific `userLogin`, `email`, `siteId` and `expiryInDays` parameters value then read back the created user to assert that all the supplied parameters were applied as expected. This will protect against future regressions.

* [x] [Functional review done](https://developer.matomo.org/guides/pull-request-eviews#functional-review-done)
* [x] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [x] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [x] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [x] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [x] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [x] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [x] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [x] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [x] Existing documentation updated if needed

File: plugins/UsersManager/tests/Integration/APITest.php [hunk 1]
```diff
@@ -1232,6 +1233,37 @@ public function test_setSuperUserAccess_failsIfCurrentPasswordIsIncorrect()
...
+          'UsersManager.inviteUser',
+          [
+            'userLogin'    => "testInviteUser",
+            'email'        => "testInviteUser@example.com",
+            'expiryInDays' => 7
+          ]
+        );
+    }
+
+    public function testInviteUserInitialIdSite()
```
Comment (peterhashair) [inline plugins/UsersManager/tests/Integration/APITest.php]:
```suggestion
    public function testInviteUserInitialIdSiteMissingWithSuperAdminAccess()
```

File: plugins/UsersManager/API.php [hunk 2]
```diff
@@ -761,9 +762,14 @@ public function inviteUser($userLogin, $email, $idSite = null, $expiryInDays = n
             $expiryInDays = Config\GeneralConfig::getConfigValue('default_invite_user_token_expiry_days');
         }
 
-        $idSite = $idSite === null ? null : intval($idSite);
+        if (empty($initialIdSite)) {
+            throw new \Exception(Piwik::translate("UsersManager_AddUserNoInitialAccessError"));
+        }else{
```
Comment (bx80) [inline plugins/UsersManager/API.php]:
```suggestion
        } else {
```

**Review by bx80 — APPROVED** [review]
I've taken a look through the other API methods added by this PR and they all seem to have appropriate unit tests. It's now not possible to invite a user without explicitly specifying a site, either from the UI or via the API :+1: 

* [x] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)

If the build tests all pass then this should be good to go.

**Comment by peterhashair** [comment]
Our Team finally agree to throw an exception on InviteUser API when initSiteID is empty, but there is no change to addUser API.

**Pull request merged**